src/renderer.ts

```diff
@@ -156,6 +158,632 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+    const celldimension = cell_dim(total_width, bins);
+    var color: any = d3.scale.linear<number, string>();
+    color = (min < 0) ? color.domain([min, 0, max]).range(colorrange)
+      : color.domain([min, max]).range([colorrange[1], colorrange[2]]);
+    const padding = context.option('rowPadding', 1);
+    return {
+      template: `<div class="heatmapcell" style="top:${padding}px;">
+                                   </div>`,
+      update: (n: HTMLDivElement, d: IDataRow, i: number) => {
+        const g = d3.select(n);
+        const div = g.selectAll('div').data(col.getValue(d.v, i));
+        div.enter().append('div');
+        div
+          .style({
+            'width': celldimension + 'px',
+            'background-color': color,
+            'height': context.rowHeight(i) + 'px',
+            'left': (d, i) => (i === null || 0) ? 0 + 'px' : (i * celldimension) + 'px'
+          });
+
+
+      }
+    };
+  }
+
+  createCanvas(col: Column, context: ICanvasRenderContext): ICanvasCellRenderer {
+    function cell_dim(total, cells) {
+      return (total / cells);
+    }
+
+    const total_width = col.getWidth();
+    const bins = (<any>col.desc).datalength;
+    const defaultdomain = [0, 100];
+    var domain = (<any> col.desc).domain;
+    domain = (domain === undefined || null) ? defaultdomain : domain;
+    const min = domain[0], max = domain[1];
+    var colorrange = (<any> col.desc).colorrange;
+    const defaultcolor = ['blue', 'red'];
+    colorrange = (colorrange === undefined || null) ? defaultcolor : colorrange;
+    const celldimension = cell_dim(total_width, bins);
+    var color: any = d3.scale.linear<number, string>();
+    color = (min < 0) ? color.domain([min, 0, max]).range([colorrange[0], 'white', colorrange[1]])
+      : color.domain([min, max]).range(['white', colorrange[1]]);
+    const padding = context.option('rowPadding', 1);
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+      var data = col.getValue(d.v, i);
+      data.forEach(function (d, i) {
+        var x = (i === null || 0) ? 0 : (i * celldimension);
+        ctx.beginPath();
+        ctx.fillStyle = color(d);
+        ctx.fillRect(x, padding, celldimension, context.rowHeight(i));
+      });
+    };
+  }
+
+}
+
+
+class SparklineCellRenderer extends DefaultCellRenderer {
+
+  createSVG(col: Column, context: IDOMRenderContext): ISVGCellRenderer {
+
+    const min = (<any> col.desc).domain[0];
+    const max = (<any> col.desc).domain[1];
+    const bins = (<any> col.desc).datalength;
+    var x: any = d3.scale.linear().domain([0, bins - 1]).range([0, col.getWidth()]);
+    var y: any = y = d3.scale.linear().domain([min, max]);
+    var line = d3.svg.line<number>();
+
+    return {
+
+      template: `<g class="sparklinecell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const g = d3.select(n);
+        const path = g.selectAll('path').data([<any>col.getValue(d.v, i)]);
```

Comment:
d3 binding for a single data item is not necessary. why not creating the path item in the template already or just the path item without any `g` wrapper